pkg/asset/installconfig/aws/dns.go

```diff
@@ -95,36 +79,31 @@ func GetBaseDomain() (zone *Zone, err error) {
 }
 
 // GetPublicZone returns a public route53 zone that matches the name.
-func GetPublicZone(sess *session.Session, name string) (*route53.HostedZone, error) {
-	var res *route53.HostedZone
-	f := func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
-		for idx, zone := range resp.HostedZones {
-			if zone.Config != nil && !aws.BoolValue(zone.Config.PrivateZone) && strings.TrimSuffix(aws.StringValue(zone.Name), ".") == strings.TrimSuffix(name, ".") {
-				res = resp.HostedZones[idx]
-				return false
-			}
-		}
-		return !lastPage
+func GetPublicZone(session *session.Session, zoneName string) (*route53.HostedZone, error) {
+	client, err := NewClientWithSession(session)
+	if err != nil {
+		return nil, err
 	}
 
-	client := route53.New(sess)
-	if err := client.ListHostedZonesPages(&route53.ListHostedZonesInput{}, f); err != nil {
-		return nil, errors.Wrap(err, "listing hosted zones")
-	}
-	if res == nil {
-		return nil, errors.Errorf("No public route53 zone found matching name %q", name)
+	zone, err := client.GetPublicHostedZone(zoneName)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get hosted zone")
 	}
 
-	return res, nil
+	return zone, nil
 }
 
 // GetPublicZoneByID returns a public route53 zone from the zone ID
-func GetPublicZoneByID(sess *session.Session, zoneID string) (*route53.HostedZone, error) {
-	client := route53.New(sess)
-	zoneOutput, err := client.GetHostedZone(&route53.GetHostedZoneInput{Id: aws.String(zoneID)})
+func GetPublicZoneByID(session *session.Session, zoneID string) (*route53.HostedZone, error) {
+	client, err := NewClientWithSession(session)
+	if err != nil {
+		return nil, err
+	}
+
+	zone, err := client.GetPublicHostedZoneByID(zoneID)
 	if err != nil {
-		return nil, errors.Wrap(err, "getting hosted zone")
+		return nil, errors.Wrap(err, "could not get hosted zone")
 	}
 
-	return zoneOutput.HostedZone, nil
+	return zone, nil
 }
```

Comment:
do we need this function anymore? why not just

    create a client, and
    call the client.GetPublicHostedZoneByID(zoneID) ?